database/meta.go

```diff
@@ -64,98 +63,61 @@ func NewMetaDB(config *config.Config, dbHandle *mgo.Session,
 	return metaDB
 }
 
-//EnsureRollingSettingsMatch ensures that the rolling database settings in the
-//metadb match the provided arguments if the database settings exist at all
-func (m *MetaDB) EnsureRollingSettingsMatch(db string, numchunks int) error {
-	m.lock.Lock()
-	defer m.lock.Unlock()
-	ssn := m.dbHandle.Copy()
-	defer ssn.Close()
-
+//GetRollingSettings gets the current rolling settings
+func (m *MetaDB) GetRollingSettings(db string) (exists bool, isRolling bool, currChunk int, totalChunks int, err error) {
 	// pull down dataset record from metadatabase
-	var result DBMetaInfo
-	err := ssn.DB(m.config.S.MongoDB.MetaDB).C(m.config.T.Meta.DatabasesTable).Find(bson.M{"name": db}).One(&result)
-
+	result, err := m.GetDBMetaInfo(db)
 	if err != nil && err != mgo.ErrNotFound {
-		return err
+		return
 	}
 
-	// if dataset doesn't exist yet
-	if err == mgo.ErrNotFound {
-		return nil
+	if err != mgo.ErrNotFound {
+		exists = true
 	}
-
-	// if dataset is an already existing dataset marked as rolling = true
-	if result.Rolling {
-		// make sure number of chunks matches the one that's on file for that dataset
-		if result.TotalChunks != numchunks {
-			return fmt.Errorf("The total chunk size for existing rolling dataset [ "+db+" ] is set to [ %d ] and cannot be changed unless the dataset is deleted and recreated.", result.TotalChunks)
-		}
-	}
-
-	return nil
+	err = nil
+	isRolling = result.Rolling
+	currChunk = result.CurrentChunk
+	totalChunks = result.TotalChunks
+	return
 }
 
 //SetRollingSettings ensures that a given db is marked as rolling,
 //ensures that total_chunks matches numchunks, and sets the current_chunk to chunk.
-func (m *MetaDB) SetRollingSettings(db string, numchunks int, chunk int) error {
+func (m *MetaDB) SetRollingSettings(db string, chunk int, numchunks int) error {
```

Comment:
This function was updated to allow changing the rolling state and the total number of chunks of an existing database.